### ISSUE TITLE
rails server --port=XXXX listening on 3000

### ISSUE THREAD

**Issue by francescor** [rails/rails#30588]
### Steps to reproduce

```
$ rails server --port=5000
=> Booting Puma
=> Rails 5.1.3 application starting in development on http://localhost:5000
=> Run `rails server -h` for more startup options
Puma starting in single mode...
* Version 3.9.1 (ruby 2.3.2-p217), codename: Private Caller
* Min threads: 5, max threads: 5
* Environment: development
* Listening on tcp://0.0.0.0:3000
```

While the following works as expected
```
$ rails server --port 5000
$ rails server -p 5000
```
### System configuration

```
Ruby rvm
$ ruby -v
ruby 2.3.2p217 (2016-11-15 revision 56796) [x86_64-linux]
rails -v
Rails 5.1.3
$ lsb_release -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	CentOS
Description:	CentOS Linux release 7.3.1611 (Core) 
Release:	7.3.1611
```

**Comment by ccasabona** [comment]
You can specify the port as a constant (meaning ALL CAPS) if you place the specification before the rails server command:

PORT=5000 rails server

If you put PORT=5000 after the rails server command, active support thinks you're trying to load a file.

**Comment by y-yagi** [comment]
This has already been fixed in master(#29146).
However it seems that it was not backported. I created a backport PR #30597.

**Comment by y-yagi** [comment]
This issue fixed by #30597.
